app/models/task.rb

```diff
@@ -19,10 +19,27 @@ class Task < ApplicationRecord
   belongs_to :supplier
 
   has_many :submissions, dependent: :nullify
-  has_one :latest_submission, -> { order(created_at: :desc) }, inverse_of: :task, class_name: 'Submission'
 
   delegate :name, to: :supplier, prefix: true
 
+  completed_or_latest_scope = lambda do
+    where("
+        CASE WHEN
+          EXISTS(
+            SELECT id
```

Comment:
I think you can do `SELECT 1` here if you're just on an existence check

Reply:
Nice idea. Doing.